scripts/redirectURLParser.js

```diff
@@ -30,7 +30,9 @@ function redirectURLParser(location) {
   }
 
   // invalid lang but with a valid path.
-  if (!langTest && Object.keys(pages).indexOf(pathname) !== -1) {
+  if (!langTest && Object.keys(pages).indexOf(redirectQuery) !== -1) {
```

Comment:
Interesting!

I think the result from `i18n.urlOverrideLang(redirectQuery);` is incorrect, and should maybe be fixed internally.

I took a look at the internals of that function, and found this line:

`var pathname = localPath.split('/')[2] || localeCode;`

Seems wrong to me. It's assuming anything with with two levels for the path means the first one must be a locale, and that's not true anymore. (at one point we just had only one level deep for all our pages.)

I would prefer if `urlOverrideLang` was fixed to handle multiple levels int he path. Thoughts?